File: rust/561.array-partition.rs
/*
 * @lc app=leetcode id=561 lang=rust
 *
 * [561] Array Partition
 */

// @lc code=start
impl Solution {
    pub fn array_pair_sum(nums: Vec<i32>) -> i32 {
        let mut nums = nums;
        nums.sort_unstable();
        nums.iter().step_by(2).sum()
    }
}
// @lc code=end

File: javascript/561.array-partition.js
/*
 * @lc app=leetcode id=561 lang=javascript
 *
 * [561] Array Partition
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var arrayPairSum = function (nums) {
  nums.sort((a, b) => a - b);
  return nums.reduce((acc, n, i) => (acc += i % 2 === 0 ? n : 0), 0);
};
// @lc code=end
